tests/components/miele/test_config_flow.py

```diff
@@ -0,0 +1,221 @@
+"""Test the Miele config flow."""
+
+from unittest.mock import patch
+
+from pymiele import OAUTH2_AUTHORIZE, OAUTH2_TOKEN
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.miele.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import CLIENT_ID
+
+from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+from tests.typing import ClientSessionGenerator
+
+REDIRECT_URL = "https://example.com/auth/external/callback"
+
+
+@pytest.mark.usefixtures("current_request_with_host")
+async def test_full_flow(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    aioclient_mock: AiohttpClientMocker,
+    setup_credentials: None,
```

Comment:
in theory you can add setup_credentials to autouse = True as you will always need it